### PR TITLE
Implement remote data pull synchronization

### PR DESCRIPTION
**Pull request by darinkrauss**
- LOOP-19 Refactor RemoteData service to pull data from sources
- LOOP-342 Remove carb entry upload status icon

Specifically:
- Update ServicesManager and RemoteDataServicesManager to support pull synchronization
- Update DeviceDataManager to support store delegation
- Add SettingsStore and DoseDecisionStore to LoopDataManager
- Upload all data when new remote data service is added

### CODE REVIEW

File: Loop/Managers/RemoteDataServicesManager.swift [hunk 1]
```diff
@@ -6,165 +6,387 @@
 //  Copyright © 2016 Nathan Racklyeft. All rights reserved.
 //
 
+import os.log
 import Foundation
 import LoopKit
 
-final class RemoteDataServicesManager: CarbStoreSyncDelegate {
+protocol RemoteDataServicesManagerDelegate: AnyObject {
```
Comment (ps2) [inline Loop/Managers/RemoteDataServicesManager.swift:13]: As discussed before, a delegate protocol for providing dependencies is unneeded indirection at this level. Created https://tidepool.atlassian.net/browse/LOOP-969 to track straightening this out.

**Review by ps2 — APPROVED** [review]
LGTM! Did a quick review, as most of this has already been reviewed.

File: Loop/Managers/RemoteDataServicesManager.swift (hunk 1 above)
Comment (darinkrauss) [reply Loop/Managers/RemoteDataServicesManager.swift:13]: Thanks!

**Review by nhamming — APPROVED** [review]
LGTM

**Pull request merged**